includes/base/controls-stack.php

```diff
@@ -393,6 +393,12 @@ public function add_control( $id, array $args, $options = [] ) {
 			'position' => null,
 		];
 
+		$should_optimize_controls = $this->should_optimize_controls();
+
+		if ( $should_optimize_controls && ! empty( $args['type'] ) && in_array( $args['type'], [ Controls_Manager::RAW_HTML, Controls_Manager::DIVIDER, Controls_Manager::HEADING, Controls_Manager::BUTTON ] ) ) {
```

Comment:
`if ( $should_optimize_controls && ! empty( $args['type'] ) && in_array( $args['type'], [ Controls_Manager::RAW_HTML, Controls_Manager::DIVIDER, Controls_Manager::HEADING, Controls_Manager::BUTTON ] ) ) {`

Maybe we can extract this entire check into a helper function as well?

`if ( $this->control_can_be_skipped() ) {` <-- Not sure about the naming, but it would help improve readability.